### ISSUE TITLE
Failing React Native e2e test on `main`

### ISSUE THREAD

**Issue by nicolo-ribaudo** [babel/babel#16024]
https://github.com/babel/babel/actions/runs/6452922682/job/17515822084

```
npx react-native init rnbabel
npm WARN exec The following package was not found and will be installed: react-native@0.72.5
npm WARN deprecated @babel/plugin-proposal-class-properties@7.18.6: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-class-properties instead.
npm WARN deprecated @babel/plugin-proposal-optional-chaining@7.21.0: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-optional-chaining instead.
npm WARN deprecated @babel/plugin-proposal-nullish-coalescing-operator@7.18.6: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-nullish-coalescing-operator instead.
npm WARN deprecated @babel/plugin-proposal-async-generator-functions@7.20.7: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-async-generator-functions instead.
npm WARN deprecated @babel/plugin-proposal-optional-catch-binding@7.18.6: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-optional-catch-binding instead.
npm WARN deprecated @babel/plugin-proposal-object-rest-spread@7.20.7: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-object-rest-spread instead.
npm WARN deprecated @babel/plugin-proposal-numeric-separator@7.18.6: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-numeric-separator instead.
npm WARN deprecated uglify-es@3.3.9: support for ECMAScript is superseded by `uglify-js` as of v3.13.0
                                                          
               ######                ######               
             ###     ####        ####     ###             
            ##          ###    ###          ##            
            ##             ####             ##            
            ##             ####             ##            
            ##           ##    ##           ##            
            ##         ###      ###         ##            
             ##  ########################  ##             
          ######    ###            ###    ######          
      ###     ##    ##              ##    ##     ###      
   ###         ## ###      ####      ### ##         ###   
  ##           ####      ########      ####           ##  
 ##             ###     ##########     ###             ## 
  ##           ####      ########      ####           ##  
   ###         ## ###      ####      ### ##         ###   
      ###     ##    ##              ##    ##     ###      
          ######    ###            ###    ######          
             ##  ########################  ##             
            ##         ###      ###         ##            
            ##           ##    ##           ##            
            ##             ####             ##            
            ##             ####             ##            
            ##          ###    ###          ##            
             ###     ####        ####     ###             
               ######                ######               
                                                          

                  Welcome to React Native!                
                 Learn once, write anywhere               

- Downloading template
✔ Downloading template
- Copying template
/home/runner/.npm/_npx/7930a8670f922cdb/node_modules/graceful-fs/polyfills.js:265
        if (!chownErOk(er)) throw er
                            ^

Error: ENOENT: no such file or directory, chmod '/tmp/rnbabel/ios/HelloWorld/Images.xcassets/AppIcon.appiconset/Contents.json'
    at Object.chmodSync (node:fs:1961:3)
    at Object.chmodSync (/home/runner/.npm/_npx/7930a8670f922cdb/node_modules/graceful-fs/polyfills.js:263:21)
    at ReadStream.<anonymous> (/home/runner/.npm/_npx/7930a8670f922cdb/node_modules/@react-native-community/cli/build/tools/copyFiles.js:83:19)
    at ReadStream.emit (node:events:514:28)
    at emitCloseNT (node:internal/streams/destroy:132:10)
    at process.processTicksAndRejections (node:internal/process/task_queues:81:21) {
  errno: -2,
  syscall: 'chmod',
  code: 'ENOENT',
  path: '/tmp/rnbabel/ios/HelloWorld/Images.xcassets/AppIcon.appiconset/Contents.json'
}

Node.js v20.8.0
```

Maybe @cortinico you might know what causes `npx react-native init rnbabel` in a `/tmp` directory to fail in GitHub actions? :) It started failing on September 29th.

**Comment by JLHwung** [comment]
The upstream fix https://github.com/react-native-community/cli/pull/2096 has been released. Maybe `react-native` has not yet updated the dependency of `@react-native-community/cli`?

**Comment by cortinico** [comment]
> The upstream fix [react-native-community/cli#2096](https://github.com/react-native-community/cli/pull/2096) has been released. Maybe `react-native` has not yet updated the dependency of `@react-native-community/cli`?

That's the exact answer. We've fixed it in the CLI but haven't bumped it yet in React Native + we'll have to release a new version of React Native. It's odd this started happening out of the blue though

**Comment by nicolo-ribaudo** [comment]
> It's odd this started happening out of the blue though

We don't have lockfiles in our e2e tests to make sure to test the latest versions, so that's probably the reason.

**Comment by cortinico** [comment]
> We don't have lockfiles in our e2e tests to make sure to test the latest versions, so that's probably the reason.

Still `npx react-native init` is using the latest stable (0.72.0) version which was published 4 months ago (there were patches though, but they should have not affected this)

**Comment by JLHwung** [comment]
> That's the exact answer. We've fixed it in the CLI but haven't bumped it yet in React Native + we'll have to release a new version of React Native. It's odd this started happening out of the blue though

Since the issue is a racing condition and in our case we are writing to `/tmp`. I guess recently GitHub CI somehow throttled IO on `/tmp` so that now it is more likely that when file's read stream is closed, its copy is not yet written to the destination.

> Still npx react-native init is using the latest stable (0.72.0) version which was published 4 months ago

The upstream patch is released on 12.0.0-alpha only, does `react-native` 0.72 use `@react-native-community/cli` 11? If so we should consider backporting the fix if a new major release of `react-native` is not expected soon.

**Comment by cortinico** [comment]
> The upstream patch is released on 12.0.0-alpha only, does `react-native` 0.72 use `@react-native-community/cli` 11? If so we should consider backporting the fix if a new major release of `react-native` is not expected soon.

Yup I believe we'll backport this down to other versions as well (cc @thymikee)
